### ISSUE TITLE
Copy link to book/note/... for external program

### ISSUE THREAD

**Issue by DimitriyPro** [TriliumNext/Trilium#1012]
Hi team,
I've just find Trilium notes and it's best program for note taking, I suppose. So, I can't find an easy way to copy a link on book/note/ etc. If I choose edit a link, I can copy something like this http://127.0.0.1:37840/#root/WJgHkuKn9kSy/svHxOLcYwmIs/AAsx4NSZiqc4/Hwnc96O7wlg2
But I can't get an access via this link.
For example in OneNote links looks like this:
onenote:///C:\Notes\OneNote\lib\cpp.one#build&section-id={14B3EE87-2BD1-4804-8E25-CA10FEC5352B}&page-id={363824B7-6167-4715-9EA1-EE87B5BBD0E8}&object-id={231F4F9E-BB85-01DD-2368-26AE922C4AB6}&1E
Trilium approach is more usefull, since this type of links can be used in different software like TheBrain but how to fix it?
Thank you!

**Comment by zadam** [comment]
Hello, so the reason why onenote can be opened externally is because it registers custom "onenote://" protocol handler.

This has not been implemented yet. This has been already reported but unfortunately can't find the exact issue.

**Comment by DimitriyPro** [comment]
You have all for done it - local web-server on port 37840 is more powerfull and cross-platform then OneNote solution - just return a note as web-page.

**Comment by zadam** [comment]
The problem is that this would be quite unsecure - any application running on your computer could steal your notes by just querying for the port 37840 and following the links. The better way would be the protocol handler - the link just opens the Trilium app and so data stealing is not that easy ...

**Comment by DimitriyPro** [comment]
Ok, but I hope you agree that users want usefull feature right now, migrare from discomfort notetaking software and support your project. For example, when you think about my security and don't add this feature I still use OneNote (((  I think there is three steps to solve this trouble (I hope it will be usefull for you): add option "start web-server by your own risk" -> add password request (floating window) with session -> add two-factor auth with token (file on usb or mobile app).

**Comment by zadam** [comment]
> Ok, but I hope you agree that users want usefull feature right now

Well, I would like to have all the features implemented right now as well :-)

My time is limited and I have to use it with thought. Part of that is that I don't want to implement temporary half hack/half solution when a real solution is in sight (even if it takes more time).

**Comment by gerroon** [comment]
@zadam 

Why not create a special link that can be opened in a browser? That is how Nextcloud does shares, it just creates a special link which can be opened by anyone who has that particular link.

**Comment by zadam** [comment]
What you describe would indeed work for sharing between people.

But I think this is a different use case - Dimitry is not trying to send note to somebody else, only link locally and for that reusing "share" functionality would not lead to expected result (since it would open a read only page in a browser instead of an app).

**Comment by DimitriyPro** [comment]
Yes, you are right dear @gerroon 
I mean that web-access to notes is killing feature: for private using and for share it. Custom protocol handler it is not so good way to do it (sorry for it).
For example, users can get all benefits of url instant view.
TheBrain (c) example:
![2020-05-13_9-55-35](https://user-images.githubusercontent.com/55881774/81793707-a725ce00-9501-11ea-832b-fa5e9bd6aad6.png)

OneNote with custom handler looks strange:
![2020-05-13_9-59-37](https://user-images.githubusercontent.com/55881774/81793791-c290d900-9501-11ea-8343-0b5ff3f7253f.png)

Trilium with web-access looks perfect and native:
![2020-05-13_9-59-07](https://user-images.githubusercontent.com/55881774/81793914-e48a5b80-9501-11ea-89b9-5c68dab2328b.png)

**Comment by DimitriyPro** [comment]
Obsidian implemented this feature(((
obsidian://open?vault=Test&file=%D0%9B%D0%B5%D0%B2%20%D0%A2%D0%BE%D0%BB%D1%81%D1%82%D0%BE%D0%B9
@zadam can I donate for it?

**Comment by r-nd-m** [comment]
+1 for the feature to copy note link to be used in external application and directly open specific note in Trilium application

**Comment by grotesque** [comment]
+1 for the feature

**Comment by zadam** [comment]
@meichthys yes, this is actually a CKEditor problem. Read only doesn't use CKEditor, it just renders the HTML directly, that's why it works.

**Comment by eliandoran** [comment]
Duplicate of https://github.com/TriliumNext/Trilium/issues/649.
